app/routes/events/components/JoinEventForm.js

```diff
@@ -379,7 +382,31 @@ const JoinEventForm = (props: Props) => {
                   </Link>
                 </div>
               )}
+            {!disabledForUser &&
+              event.useConsent &&
+              !hasRegisteredConsent(
+                currentUser.photoConsents,
+                getEventSemesterFromStartTime(event.startTime)
+              ) && (
+                <div className={styles.eventWarning}>
+                  <p>NB!</p>
+                  <p>
+                    Du må ta stilling til bildesamtykke for semesteret{' '}
+                    {toReadableSemester(
+                      getEventSemesterFromStartTime(event.startTime)
+                    )}{' '}
+                    for å melde deg på dette arrangement.
+                  </p>
+                  <Link to={`/users/me/`}>Gå til min profil</Link>
+                </div>
+              )}
             {formOpen &&
+              (event.useConsent
+                ? hasRegisteredConsent(
+                    currentUser.photoConsents,
+                    getEventSemesterFromStartTime(event.startTime)
+                  )
+                : true) &&
```

Comment:
If you could, I think this is at the point where it needs some refactoring into something more readable (and not in the middle of the JSX)

Reply:
✅